multiplex_immuno_processing/compute_alignment.py

```diff
@@ -0,0 +1,238 @@
+
+
+# final_z_offset = 7
+# final_y_offset = 202
+# final_x_offset = 110
+
+# TODO: save out the output version of these files here
+# save original images for alignment
+
+tifffile.imwrite("source_for_alignment.tiff", source_img_cropped)
+tifffile.imwrite("target_for_alignment.tiff", target_img)
+
+
+
+
+
+#print(f"source is {np.shape(source_aligned)}")
+#print(f"target img is of shape is {np.shape(target_aligned)}")
+# This part is just for visualization purposes
+tifffile.imwrite("target_img_round_1_aligned.tiff", target_aligned)
+tifffile.imwrite("source_img_round_2_aligned.tiff", source_aligned)
+
+final_shape = np.uint16(
+    np.asarray(
+        [
+            100,
+            1248 + 1248 / 3,
+            1848 + 1848 / 3,
+        ]
+    )
+)
+
+
+#target_img_padded = pad_target_image(target_img, z_pad_offset, y_pad_offset, x_pad_offset)
+
+
+print(f"final z alignment is {final_z_offset}")
+print(f"final y alignment is {final_y_offset}")
+print(f"final x alignment is {final_x_offset}")
+
+#alignment_offset = [final_z_offset-z_pad_offset, final_y_offset-y_pad_offset, final_x_offset-x_pad_offset]
+
+
+alignment_offset = [final_z_offset, final_x_offset-150, final_y_offset-150] #
+
+
+#alignment_offset = [-final_z_offset, final_y_offset-150 ,final_x_offset-150]
+align_shift_list = [np.asarray([0, 0, 0]), alignment_offset] # THis is because first image is the refrence image
+
+
+aligned_img_list = []
+unaligned_img_list = []
+count=0
+
+
+
+
+# align_matrix = get_align_matrix(alignment_offset)
+
+# shift_to_center_matrix = get_shift_to_center_matrix(imgstack.shape, final_shape)
+
+# print(f"align matrix is {align_matrix}")
+# print(f"shift_to_center_matrix  is {shift_to_center_matrix}")
+
+# combo = shift_to_center_matrix @ align_matrix 
+# print(f"combo is {combo}")
+
+# processed_volume = affine_transform(imgstack, np.linalg.inv(combo), output_shape=final_shape, order=0)
+
+# print(np.shape(processed_volume))
+
+#tifffile.imwrite(os.path.join(output_example_dir, f"count_{1}.tiff"), processed_volume)
+
+
+for alignment_offset, img in zip(align_shift_list, imglist):
+    print(alignment_offset)
+    align_matrix = get_align_matrix(alignment_offset)
+    print(f"align matrix is {align_matrix}")
+    shift_to_center_matrix = registration_utils.get_shift_to_center_matrix(np.shape(img), final_shape)
+    combo = shift_to_center_matrix @ align_matrix
+    print(f"shift to center matrix is {shift_to_center_matrix}")
+    print(f"combo is {combo}")
+    print(f"image stack is of shape {np.shape(img.shape)}")
+    processed_volume = registration_utils.affine_transform(img, np.linalg.inv(combo), output_shape=final_shape, order=0)
+    #print(f"img is {img}")
+    #print(f"alignment offset is {alignment_offset}")
+    tifffile.imwrite(os.path.join(output_example_dir, f"count_{count}.tiff"), processed_volume)
```

Comment:
Lets try to use aicsimageio for all file writing tasks

Reply:
We don't actually need this script in the workflow with the new jinja templates. Will remove this